clients/ui/frontend/src/app/pages/modelCatalog/utils/hardwareConfigurationFilterUtils.ts

```diff
@@ -110,19 +111,14 @@ export const filterHardwareConfigurationArtifacts = (
       }
     }
 
-    // Max Latency Filter - use provided config or fall back to default
-    const maxLatencyFilter = filterState[ModelCatalogNumberFilterKey.MAX_LATENCY];
-    if (maxLatencyFilter !== undefined) {
-      const baseConfig = latencyConfig || {
-        metric: LatencyMetric.TTFT,
-        percentile: LatencyPercentile.Mean,
-      };
-
-      // Create the full config with the current filter value
-      const fullConfig: LatencyFilterConfig = { ...baseConfig, value: maxLatencyFilter };
-
-      if (!applyMaxLatencyFilter(artifact, fullConfig)) {
-        return false;
+    // Max Latency Filter - check for any active latency field
+    for (const fieldName of LATENCY_METRIC_FIELD_NAMES) {
+      const filterValue = filterState[fieldName];
+      if (filterValue !== undefined && typeof filterValue === 'number') {
+        const latencyValue = getDoubleValue(artifact.customProperties, fieldName);
+        if (latencyValue > filterValue) {
+          return false;
+        }
       }
     }
```

Comment:
Same here, you can derive everything from the separate metric and percentile enums
```suggestion
    // Max Latency Filter - check for any active latency field
    for (const metric of Object.values(LatencyMetric)) {
      for (const percentile of Object.values(LatencyPercentile)) {
        const fieldName = getLatencyFieldName(metric, percentile);
        const filterValue = filterState[fieldName];
        if (filterValue !== undefined && typeof filterValue === 'number') {
          const latencyValue = getDoubleValue(artifact.customProperties, fieldName);
          if (latencyValue > filterValue) {
            return false;
          }
        }
      }
    }
```